smithy-cli/src/main/java/software/amazon/smithy/cli/commands/LockFile.java

```diff
@@ -0,0 +1,246 @@
+                throw new CliError("Resolved artifact `" + artifact + "` does not match pinned artifact `" + pin + "`");
+            }
+        }
+    }
+
+    public static LockFile.Builder builder() {
+        return new LockFile.Builder();
+    }
+
+    /**
+     * @return Gets the configuration hash for the lockfile
+     */
+    public int getConfigHash() {
+        return configHash;
+    }
+
+    /**
+     * @return Gets the list of pinned artifacts.
+     */
+    public Set<String> getDependencyCoordinateSet() {
+        return artifacts.keySet();
+    }
+
+    /**
+     * @return Gets the version of lockfile.
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * @return Gets the pinned repositories in the lockfile.
+     */
+    public Set<String> getRepositories() {
+        return repositories;
+    }
+
+
+    @Override
+    public LockFile.Builder toBuilder() {
+        return builder().repositories(repositories)
+                .artifacts(artifacts)
+                .version(version)
+                .configHash(configHash);
+    }
+
+    public static final class Builder implements SmithyBuilder<LockFile> {
+        private final BuilderRef<Map<String, PinnedArtifact>> artifacts = BuilderRef.forOrderedMap();
+        private final BuilderRef<Set<String>> repositories = BuilderRef.forOrderedSet();
+        private String version = DEFAULT_VERSION;
+        private int configHash;
+
+        public LockFile.Builder artifacts(Map<String, PinnedArtifact> artifacts) {
+            this.artifacts.clear();
+            this.artifacts.get().putAll(artifacts);
+            return this;
+        }
+
+        public LockFile.Builder artifacts(Collection<ResolvedArtifact> artifacts) {
+            this.artifacts.clear();
+            for (ResolvedArtifact artifact : artifacts) {
+                this.artifacts.get().put(artifact.getCoordinates(), PinnedArtifact.from(artifact));
+            }
+            return this;
+        }
+
+        public LockFile.Builder artifact(ResolvedArtifact artifact) {
+            this.artifacts.get().put(artifact.getCoordinates(), PinnedArtifact.from(artifact));
+            return this;
+        }
+
+        public LockFile.Builder artifact(PinnedArtifact artifact) {
+            this.artifacts.get().put(artifact.coordinates, artifact);
+            return this;
+        }
+
+        public LockFile.Builder repositories(Set<MavenRepository> repositories) {
+            this.repositories.clear();
+            for (MavenRepository repository : repositories) {
+                this.repositories.get().add(repository.getUrl());
+            }
+            return this;
+        }
+
+        public LockFile.Builder repositories(Collection<String> repositories) {
+            this.repositories.clear();
+            this.repositories.get().addAll(repositories);
+            return this;
+        }
+
+        public LockFile.Builder repository(MavenRepository repository) {
```

Comment:
If you aren't using these singular methods (repository() / artifact()), then consider removing.